packages/@react-aria/utils/src/useSyntheticBlurEvent.ts

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {FocusEvent as ReactFocusEvent, useCallback, useRef} from 'react';
+import {useLayoutEffect} from './useLayoutEffect';
+
+class SyntheticBlurEvent implements ReactFocusEvent {
+  nativeEvent: FocusEvent;
+  target: Element;
+  currentTarget: Element;
+  relatedTarget: Element;
+  bubbles: boolean;
+  cancelable: boolean;
+  defaultPrevented: boolean;
+  eventPhase: number;
+  isTrusted: boolean;
+  timeStamp: number;
+  type: string;
+
+  constructor(type: string, nativeEvent: FocusEvent) {
+    this.nativeEvent = nativeEvent;
+    this.target = nativeEvent.target as Element;
+    this.currentTarget = nativeEvent.currentTarget as Element;
+    this.relatedTarget = nativeEvent.relatedTarget as Element;
+    this.bubbles = nativeEvent.bubbles;
+    this.cancelable = nativeEvent.cancelable;
+    this.defaultPrevented = nativeEvent.defaultPrevented;
+    this.eventPhase = nativeEvent.eventPhase;
+    this.isTrusted = nativeEvent.isTrusted;
+    this.timeStamp = nativeEvent.timeStamp;
+    this.type = type;
+  }
+
+  isDefaultPrevented(): boolean {
+    return this.nativeEvent.defaultPrevented;
+  }
+
+  preventDefault(): void {
+    this.defaultPrevented = true;
+    this.nativeEvent.preventDefault();
+  }
+
+  stopPropagation(): void {
+    this.nativeEvent.stopPropagation();
+    this.isPropagationStopped = () => true;
+  }
+
+  isPropagationStopped(): boolean {
+    return false;
+  }
+
+  persist() {}
+}
+
+export function useSyntheticBlurEvent(onBlur: (e: ReactFocusEvent) => void) {
+  let stateRef = useRef({
+    isFocused: false,
+    onBlur,
+    observer: null as MutationObserver,
+    target: null as HTMLElement
+  });
+  stateRef.current.onBlur = onBlur;
+
+  // Clean up MutationObserver on unmount. See below.
+  // eslint-disable-next-line arrow-body-style
+  useLayoutEffect(() => {
+    const state = stateRef.current;
+    return () => {
+      if (stateRef.current.target) {
+        stateRef.current.target.removeEventListener('focusout', onBlurHandler);
+      }
```

Comment:
Wanna make sure I understand the changes from the old useSyntheticBlurEvent:
this is in case the target (focused element) is unmounted -> mutation observer will dispatch a blur -> we need to clean up the blur handler from the target to avoid a second blur since we have a second synthetic blur event handler from `useFocus` (this second one is unavoidable the element may not be in a FocusScope)

Reply:
not quite, this is so that elements (ie FocusScopes) listening for blur on something inside them, don't try to handle a blur after they've been unmounted. the case we care about right is:
```
<FocusScope contains><button with focus /></FocusScope>
```
where the button is unmounted, but not the FocusScope which is watching for blur inside its subtree

this cleanup, handles this case:
```
<FocusScope contains>
  <FocusScope contains>
    <button with focus />
  </FocusScope>
</FocusScope>
```
Where the whole inner focus scope plus button is unmounted. we don't want that inner one to handle the blur we're about to fire

Reply:
Ah ok, I see now, this is the unmount cleanup for the FocusScope (or w/e component calls this hook) not the actual focused element at the time, gotcha.